packages/hooks/src/useObjectDependency.ts

```diff
@@ -0,0 +1,13 @@
+import { useRef } from 'react';
+import { isEqual } from 'lodash';
+
+export default function useObjectDependency<T>(object: T): T {
+  const ref = useRef<T>();
+
+  // we need isEqual for deep object comparison
+  if (ref.current === undefined || !isEqual(ref.current, object)) {
```

Comment:
consider the following
```jsx
const object = {a, b, c, d};

useEffect(() => {
  // does something with `object`
}, [a, b, c, d])
```
The linter is not smart enough to know that `object` will only have changed when `a`, `b`, `c`, and `d` have changed. It might also not be obvious to a human either for more complicated code. So you can satisfy the linter my doing
```jsx
const object = useMemo(() => ({a, b, c, d}), [a, b, c, d])

useEffect(() => {
  // does something with `object`
}, [object])
```
since the object reference shouldn't change between renders if `a`, `b`, `c`, and `d` haven't changed. Needing to do `({a, b, c, d}), [a, b, c, d])` can be annoying and verbose, so `useObjectDependency` accomplishes the same thing for you.

Reply:
`isEqual` isn't exactly the cheapest operation. I like having this in our tool belt but I think we should be cautious when grabbing for it.